lib/imports/api/authority/server/authority-manager.js

```diff
@@ -0,0 +1,204 @@
+import { Authority } from './authority';
+import { Meteor } from 'meteor/meteor';
+import { check } from 'meteor/check';
+
+export class AuthorityManager {
+  /**
+  * Manages creating Authorities for ProseMirror documents and provides communication between clients
+  * and their corresponding Authority.
+  *
+  * @param {Object} params
+  * @param {Collection} params.collection     a Collection instance that is used to store ProseMirror docs
+  * @param {Streamer}  params.streamer        a Streamer object that contains .emit() and .on() methods for streaming events with clients
+  * @constructor
+  */
+  constructor({ documentCollection, streamer }) {
+    this.documentsCollection = documentCollection;
+    this.authorities = {};
+    this.streamerServer = streamer;
+
+    this.setupClientMethods();
+  }
+
+  /**
+  * Createds Meteor Methods to allow clients to communicate with the AuthorityManager.
+  */
+  setupClientMethods() {
+    let self = this;
```

Comment:
I was originally planning on just using Streamer for all communication, but it turned out it's difficult to use streams to only communicate with a single client. So for everything that the client communicates with the server, Methods are used. This is also nice because Meteor Methods automatically retry if disconnected. Streamer is now only used for server to notify clients that there are new steps.